daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseOutputStreamContentHandler.scala

```diff
@@ -107,6 +208,36 @@ class DaffodilParseOutputStreamContentHandler(out: OutputStream, pretty: Boolean
     outputNewlineStack.push(false)
   }
 
+  private def outputTagName(uri: String, localName: String, qName: String, atts: Maybe[Attributes] = Nope): Unit = {
+    val tagName = {
+      if (qName.nonEmpty) {
+        qName
+      } else {
+        // use attributes
+        lazy val attrs = atts.get
+        lazy val in = attrs.getIndex(uri, localName)
+        if (atts.nonEmpty && in >= 0) {
+          val qn = attrs.getQName(in)
+          qn
```

Comment:
This feels wrong, or I don't understand what's going on here. So in the case where qname is empty, the uri is something like ``http://example.com`` and localName is something like ``foo``, right? And we want to figure out the qname (i.e. prefix + localName) to output for this element tag.

But doesn't that mean this code is looking for an attribute with name "foo" and namespace "https://example.com" and has the value of the qname? So is this expecting an attribute that looks like this to existt: ``{https://example.com}foo="ex:foo"``? Does that kind of attribute actually exist, or am I reading this wrong?

Reply:
You were not reading it wrong, this was wrong. Fixed now.